src/main/java/com/google/sps/servlets/EditRecipeServlet.java

```diff
@@ -68,16 +69,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       }
 
       recipeEntity.setProperty("name", name);
-      recipeEntity.setProperty("tags", tags);
       recipeEntity.setProperty("description", description);
+      recipeEntity.setProperty("imageBlobKey", imageBlobKey);
       recipeEntity.setProperty("ingredients", ingredients);
       recipeEntity.setProperty("steps", steps);
-      recipeEntity.setProperty("imageBlobKey", (String) recipeEntity.getProperty("imageBlobKey"));
+      recipeEntity.setProperty("tags", tags);
+
+      if (privacy.equals("public")) {
+        recipeEntity.setProperty("published", true);
+      }
+      if (privacy.equals("private")) {
+        recipeEntity.setProperty("published", false);
+      }
 
       datastore.put(recipeEntity);
     } catch (EntityNotFoundException e) {
-      // in normal circumstances, this won't happen bc user has not access to id
-      System.out.println("entity not found exception");
+      System.out.println("EditRecipeServlet: Private recipe entity not found with saved recipe id. This should never happen.");
```

Comment:
I tried to automatically reload, but it leads to issues right now since every time I refresh it asks me to log in again
my goal is to reload just the divs but idk if I'll get that to work